### PR TITLE
adjusted the project code and configuration to run in the new enigmampc/docker-environment

### PR DESCRIPTION
**Pull request by reuvenpo**
This PR adds support for running Salad inside enimgampc/docker-environment.
Please check out the PR at enigmampc/docker-environment#19 to see what i'm doing with all these changes. I'm still going to polish some stuff here, but any input is obviously welcome :)
I definitely intend on making sure that i didn't break the frontend service, but i haven't done that yet.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,23 +6,32 @@
...
-    "test": "yarn dc migrate && mocha --exit ./test/api.test.js",
-    "dc": "node ./node_modules/@enigmampc/discovery-cli/src/index.js"
+    "dc": "discovery",
+    "configure": "scripts/configure.sh",
+    "configure-sw": "SGX_MODE=SW scripts/configure.sh",
+    "start": "yarn dc start",
+    "stop": "yarn dc stop",
+    "compile": "yarn dc compile",
+    "test": "yarn dc compile && mocha --exit ./test/api.test.js",
+    "migrate": "yarn dc compile && yarn dc migrate",
```
Comment (lacabra) [inline package.json]: The latest version of `discovery-cli` enforces that `compile` is always run before `migrate` so `yarn dc migrate` suffices here.

File: package.json [hunk 2]
```diff
@@ -6,23 +6,32 @@
   "directories": {
     "test": "test"
   },
+  "devDependencies": {
+    "@enigmampc/discovery-cli": "0.1.7",
```
Comment (lacabra) [inline package.json]: latest is now `0.1.8`

File: package.json (hunk 1 above)
Comment (reuvenpo) [reply package.json]: :+1:

File: package.json (hunk 2 above)
Comment (reuvenpo) [reply package.json]: :+1:

File: frontend/src/components/App.js [hunk 3]
```diff
@@ -42,8 +40,8 @@ class App extends Component {
         try {
             const web3 = await getWeb3();
             this.props.initializeWeb3(web3);
-            // TODO: Bootstrap the operator url during build
-            const salad = new CoinjoinClient( undefined, web3 );
+            const protocolHost = window.location.protocol + '//' + window.location.hostname;
```
Comment (fredfortier) [inline frontend/src/components/App.js:43]: We should define an endpoint for the operator api (e.g. http://host/api). The frontend should be in the domain root. A reverse proxy (the nginx service) should rewrite the api route on port 80 to the operator internal endpoint.

**Comment by fredfortier** [comment]
Looks good expect for the one conversation above.

File: frontend/src/components/App.js (hunk 3 above)
Comment (reuvenpo) [reply frontend/src/components/App.js:43]: I don't really know how to set up and configure nginx. do you think you could do it as a PR to this branch? (or, just a commit on this branch?)

Comment (fredfortier) [reply frontend/src/components/App.js:43]: I think that it can be a separate PR. We just need to keep track of it. I can take care of it.

**Comment by lacabra** [comment]
@reuvenpo In order to run this properly in the new `docker-environment` you need to push new vresions of the `docker-compose.hw.yml` and `docker-compose.sw.yml`, which in the case of `docker-environment` are these two files:

- https://github.com/enigmampc/docker-environment/blob/develop/docker-compose.yml
- https://github.com/enigmampc/docker-environment/blob/develop/docker-compose.hw.yml

As right now it's using the older network, and things are not working properly.

**Comment by reuvenpo** [comment]
> @reuvenpo In order to run this properly in the new `docker-environment` you need to push new vresions of the `docker-compose.hw.yml` and `docker-compose.sw.yml`, which in the case of `docker-environment` are these two files:
> 
>     * https://github.com/enigmampc/docker-environment/blob/develop/docker-compose.yml
> 
>     * https://github.com/enigmampc/docker-environment/blob/develop/docker-compose.hw.yml
> 
> 
> As right now it's using the older network, and things are not working properly.

as discussed privately, this is irrelevant.

File: package.json [hunk 4]
```diff
@@ -6,23 +6,33 @@
...
     "run-operator": "node ./operator/src/server.js",
-    "test": "yarn dc migrate && mocha --exit ./test/api.test.js",
-    "dc": "node ./node_modules/@enigmampc/discovery-cli/src/index.js"
+    "dc": "discovery",
+    "configure": "scripts/configure.sh",
+    "configure-sw": "SGX_MODE=SW scripts/configure.sh",
+    "start": "yarn dc start",
+    "stop": "yarn dc stop",
+    "compile": "yarn dc compile",
+    "test": "yarn dc compile && mocha --exit ./test/api.test.js",
```
Comment (fredfortier) [inline package.json]: Why did you change the workflow here? `migrate` is the right command, not compile. If you need a different workflow for system tests, I would recommend creating a new command.
